scripts/c-btmInference.py

```diff
@@ -0,0 +1,122 @@
+from transformers import AutoModelForCausalLM, AutoTokenizer
+import torch
+from torch.nn import PairwiseDistance
+import numpy 
+
+def load_models(model_names):
+    """
+    takes in list of model names and loads model and corresponding tokenizers
+    returns separate lists of models and tokenizers to be accessed by other functions later
+    """
+    models = []
+    tokenizers = []
+
+    for model_name in model_names:
+        model = AutoModelForCausalLM.from_pretrained(model_name)
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+
+        # Freeze the parameters of the loaded models if needed
+        model.eval()
+        for param in model.parameters():
+            param.requires_grad = False
+
+        models.append(model)
+        tokenizers.append(tokenizer)
+
+    return models, tokenizers
+
+def generateNextTokenFromExpert(prompt, tokenizer, model):
+  """
+  takes in prompt, tokenizer and model which should be passed from topKFilter
+  as topKFilter determines which domains are most relevant given the context
+  returns the predicted token of the model and its corresponding probability
+  """
+  inputs = tokenizer(prompt)
+  print(inputs['input_ids'])
+  sizeOfInputs = len(inputs['input_ids']) 
+  outputs = model(**inputs, max_new_tokens=1, 

```

Comment:
max_new_tokens not a parameter of geoptx -- how can I limit the number of tokens

Reply:
can't you do max_length ?